tutorial.lisp: insert an explanation about application-mode on built-in tutorial

pdelfino opened the pull request:
Hopefully, this fixes #1516.

Inside emacs, before commiting, I used `fill-paragraphs` and `white-space-mode`.  Depending on how `fill-paragraph` goes, it breaks the compilation because some line breaks are set in "empty" places (between strings and `(:code ....)`). I am not sure if `fill-paragraph` is the best way to keep it short. I am open to advice :wink:

I hope there will be no problem concerning whitespaces or the length of the content.

edgar-vincent commented:
That's a very good idea!
I think the tone might need to be a little more formal or abstract to blend in better with the rest of the manual. Also, I think the possessive adjective "their" should be used instead of "his".

pdelfino commented:
>   "their" should be used instead of "his".

Fixed. I am not a native speaker, so, thanks a lot! :uk:

>  think the tone might need to be a little more formal or abstract to blend in better with the rest of the manual

**On being more formal:**
Could you give me another English hint? What exactly sounds informal to you?

**On being abstract:**
I noticed that too. But I was not able to come up with something better. If we use `descibe-command` on `application-mode` we get: _Mode that forwards all keys to the renderer._

I even used that as part of my explanation. It is abstract. But I am not sure if this is clear to a lot of people . It was not clear for me the power of `application-mode` when I first read the definition. It was clear after seeing an example.

Having a look at the Common Lisp official hyper spec [documentation](http://clhs.lisp.se/Body/f_find_.htm), they provide abstract definitions _and_ examples. To be honest here, the examples are the first thing I usually read :laughing:

Maybe we could consider inserting more examples on the `tutorial`! Instead of removing this one to keep it consistent.

jmercouris commented:
Hi Pedro, I think it is more or less ready, @edgar-vincent has left the same comments I would :-). I also noticed though that there is still yet an instance of "his" in the string. Perhaps you forgot to push?

pdelfino commented:
>  also noticed though that there is still yet an instance of "his" in the string. Perhaps you forgot to push?

Oh, no. I actually forgot about it. Using `their` with the *singular* is counter-intuitive for my Brazilian brain haha. I missed the second one. My bad! It was good to learn the grammar rule abstracting the gender, thanks.

Ambrevar commented:
See https://en.wikipedia.org/wiki/Singular_they :)

Ambrevar commented on source/tutorial.lisp:
```diff
@@ -176,6 +176,16 @@ removed from the input, they are also removed from the existing bookmark.")
+   (:h3 "Application mode")
+   (:p "The command " (:code "application-mode") " forwards all keys to the
+renderer. For instance, using the default configuration of Nyxt (" (:code "web-cua-map") ") the
+key-binding " (:code "C-i") " executes a command called "(:code "autofill") ". Suppose 
```
Missing space before `(:code`

pdelfino replied:
I have implemented all of the suggestions. Thanks. 
 
 On this one, I would like to ask a question:

>  "After successfully using the keyboard shortcut for italic provided by this specific website," -> remove since it does not convey anything new.

I have already removed it but the I think the **tutorial** does not mention at any point the **fact** that to revert a mode you can execute the mode again. You can also use `disable-mode` (I did a PR to include that recently).

What do you think about adding a phrase on the **Modes** section of the tutorial to inform this?

Just to illustrate:
**Modes**
[...] _Each mode has an associated mode toggler which is a command of the same name that toggles the mode for the current buffer._ Finally, modes are enabled after they are executed and disabled if executed again. It is also possible to enable and disable modes in one or more buffers using `enable-mode` and `disable-mode` (see below).


Btw, it would be nice to have cross-references using hyperlinks on the tutorial instead of "see below". Have you guys ever thought about it?

Ambrevar commented:
This pull request removes the cl-webkit submodule.  Can you leave this out?

The addition is welcome, however it feels longish for a simple mode.
You can shorten it.  Some tips:

- `Coincidentally, ` -> `where`
- `will be ` -> `is`
- `and not " (:code "autofill") ` -> `instead`.
- `After successfully using the keyboard shortcut for italic provided by this specific website,` -> remove since it does not convey anything new.

pdelfino commented:
I have implemented all of the suggestions. Thanks. 
 
 On this one, I would like to ask a question:

>  "After successfully using the keyboard shortcut for italic provided by this specific website," -> remove since it does not convey anything new.

I have already removed it but the I think the **tutorial** does not mention at any point the **fact** that to revert a mode you can execute the mode again. You can also use `disable-mode` (I did a PR to include that recently).

What do you think about adding a phrase on the **Modes** section of the tutorial to inform this?

Just to illustrate:
**Modes**
[...] _Each mode has an associated mode toggler which is a command of the same name that toggles the mode for the current buffer._ Finally, modes are enabled after they are executed and disabled if executed again. It is also possible to enable and disable modes in one or more buffers using `enable-mode` and `disable-mode` (see below).


Btw, it would be nice to have cross-references using hyperlinks on the tutorial instead of "see below". Have you guys ever thought about it?

jmercouris commented:
Yes we've thought about it, but have not yet implemented. We would also like links for internal buffers and pagination :-D

Ambrevar commented:
The issue about cross references: https://github.com/atlas-engineer/nyxt/issues/967

Ambrevar commented on source/tutorial.lisp:
```diff
@@ -176,6 +176,15 @@ removed from the input, they are also removed from the existing bookmark.")
 list of all bookmarks."))
+   (:h3 "Application mode")
+   (:p "The command " (:code "application-mode") " forwards all keys to the
+renderer. For instance, using the default configuration of Nyxt (" (:code "web-cua-map") ") the
```
web-cua-map is not a configuration, you mean the default binding.  This sentence can be shortened then.

Ambrevar commented on source/tutorial.lisp:
```diff
@@ -176,6 +176,15 @@ removed from the input, they are also removed from the existing bookmark.")
+   (:h3 "Application mode")
+   (:p "The command " (:code "application-mode") " forwards all keys to the
+renderer. For instance, using the default configuration of Nyxt (" (:code "web-cua-map") ") the
+key-binding " (:code "C-i") " executes a command called " (:code "autofill") ". Suppose
```
Remove `a command called`.

Ambrevar commented on source/tutorial.lisp:
```diff
@@ -176,6 +176,15 @@ removed from the input, they are also removed from the existing bookmark.")
+   (:p "The command " (:code "application-mode") " forwards all keys to the
+renderer. For instance, using the default configuration of Nyxt (" (:code "web-cua-map") ") the
+key-binding " (:code "C-i") " executes a command called " (:code "autofill") ". Suppose
+a user is using their email client which has a keyboard shortcut for
```
Shorter: "which also uses `C-i' for the italic command."
